functions.php

```diff
@@ -129,6 +127,21 @@ function twentynineteen_scripts() {
 }
 add_action( 'wp_enqueue_scripts', 'twentynineteen_scripts' );
 
+/**
+ * Fix skip link focus in IE11.
+ *
+ * This does not enqueue the script because it is tiny and because it is only for IE11,
+ * thus it does not warrant having an entire dedicated blocking script being loaded.
+ *
+ * @link https://git.io/vWdr2
+ */
+function twentynineteen_skip_link_focus_fix() {
+	echo '<script>';
+	echo file_get_contents( get_template_directory() . '/js/skip-link-focus-fix.js' );
```

Comment:
`file_get_contents` is not allowed from theme check plugin https://github.com/WordPress/theme-check/issues/55
Which means this is essentially blocked from W.org

Reply:
Yes. Theme check requires to use WordPress API's wherever possible.

This change will also be required in https://github.com/Automattic/_s/pull/1323#discussion_r224329637

Reply:
Is there precedent for using WP Filesystem in themes? The `$wp_filesystem` API is located in wp-admin and is not loaded by default in WP.

Reply:
It seems not appropriate to use WP Filesystem here. WordPress uses `file_get_contents()` internally for example in `WP_Theme::get_post_templates()`. If the main concern is that `allow_url_fopen` is enabled and that a URL is passed to `file_get_contents()`, then it would seem better to define a new function in WP that is specifically for reading files from the filesystem and short-circuits if a URL is provided as the file to read.

Reply:
This is discussed many times in #themereview and it is mostly to flag questionable usage of the function. We are aware that the function is used internally, so it doesn't make sense to try to use the WP Filesystem like that. But then themes should not need to read files either.

Reply:
Why should themes not need to read files?

Reply:
As per the Theme check plugin - Not using this function is a "Warning" and not "Required" but historically this function is disallowed in the manual reviews by the reviewers for [this explaination](https://github.com/WordPress/theme-check/issues/55).

Themes do require to read from files. The most discussed use case if reading JSON files and after this PR and https://github.com/Automattic/_s/pull/1323 there will be another reason for all the themes as the default themes are role models for all the other theme developers.

Apart from `allow_url_fopen`, another discussed problem with this function is related to incorrect file permissions on some hosts. filesystem access method with access credentials can be [added in `wp-config.php`](https://codex.wordpress.org/Editing_wp-config.php#WordPress_Upgrade_Constants) in such cases, to use correct method for reading/writing to files with `WP_Filesystem`. I think @Otto42 can chime in on this if this is actually a problem or not.

If a function can be added from the core to handle these cases then it would be a great solution.

Reply:
I think that `WP_Filesystem` is only ever needed when a file needs to be written to. Reading should always be able to be done directly.

Reply:
The reason we originally banned file_get_contents was that the function was only ever used by malicious code in order to hide their malicious code. We had several examples of it in the theme directory, and no examples of it being used for good things.

The reason we left it there is because some theme developers started trying to use it to load large chunks of JSON data, like a list of google fonts, for example, then parsing that data with json_decode.. on every page load. This is slow and inefficient when you can simply pre-parse the data into a native PHP array or object and then simply include that PHP directly.

I'm not against removing the rule from theme-check, but I am generally against the notion of themes needing to read data from files in order to do something with that data, when there's simpler ways like simply including the data in the code directly. There's no reason other than organization or readability that tiny pieces of code like this can't be minified and added inline to the function, rather than being a separate file that requires additional file reads. 

Also, simply reading and outputting the file leaves the comments intact, bloating the html page output and so forth. Might as well pre-minify it and just put it right in there with wp_add_inline_script. It's tiny enough to justify that.